functions/src/aqi-calculation/buffer.ts

```diff
@@ -15,13 +15,20 @@ interface Pm25BufferElement {
  * This is a default element for the pm25 buffer. When the buffer is initialized,
  * every element is a default element. When we get an invalid reading (i.e. a
  * reading that matches that of the last reading), we put this element in the
- * buffer
+ * buffer.
+ *
+ * @readonly
+ *
+ * @remarks
```

Comment:
Maybe instead we should say to use copies of this as in `const objShallowCopy = {...Obj1};` rather than creating copies manually?

Reply:
I added a helper function that does this, lmk what you think